app/coffee/infrastructure/ExpressLocals.coffee

```diff
@@ -61,11 +61,18 @@ module.exports = (app, webRouter, apiRouter)->
 
 	webRouter.use (req, res, next)-> 
 
+		cdnBlocked = req.query.nocdn == 'true' or req.session.cdnBlocked
+
+		if cdnBlocked and !req.session.cdnBlocked?
+			ip = req.ip || req.socket?.socket?.remoteAddress || req.socket?.remoteAddress
```

Comment:
i stole it from http.coffee in metrics where do this, assumed there are some edge cases.